cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -168,32 +167,27 @@ static void gpi_load_libs(std::vector<std::string> to_load)
 
         std::string arg = *iter;
 
-        std::string lib_name;
-        std::string func_name;
-        auto it = std::find(arg.begin(), arg.end(), ':');
-        bool const has_colon = it == arg.end();
-        if (has_colon) {
-            // no colon in the string, default
-            lib_name = arg;
-            func_name = arg + "_entry_point";
-        }
-        else {
-            lib_name = std::string(arg.begin(), it);
-            func_name = std::string(it+1, arg.end());
+        auto const it = std::rfind(arg.begin(), arg.end(), ':');  // find from right since path could contain colons (Windows)
+        if (it == arg.end()) {
+            // no colon in the string
+            printf("cocotb: Error parsing GPI_EXTRA %s\n", arg.c_str());
+            exit(1);
         }
 
-        std::string full_name = "lib" + lib_name + DOT_LIB_EXT;
-        void *lib_handle = utils_dyn_open(full_name.c_str());
+        std::string const lib_name {arg.begin(), it};
+        std::string const func_name {it+1, arg.end()};
```

Comment:
```suggestion
        std::string const lib_name = arg.substr(0, idx);
        std::string const func_name = arg.substr(idx+1, std::string::npos);
```

Reply:
Private CI still seems unhappy. I guess `GPI_EXTRA` needs to be changed in the Makefiles as well?

Reply:
Log [az.log](https://github.com/cocotb/cocotb/files/5814367/az.log)

Reply:
Yeah my change requires full names. So .so/.dll extensions are necessary. The idea is whatever the user puts in GPI_EXTRA is fed directly to utils_dyn_open.